src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -1061,13 +1091,158 @@ class LicenceService(
     }
     return false
   }
+  private fun isAdditionalConditionOf14B(
+    conditionCode: String
+  ) = conditionCode == conditionCode14B
+
+  private fun replaceExpandedTextWithEndDate(
+    licenceEntity: EntityLicence,
+    expandedText: String
+  ): String {
+    var expandedText1 = expandedText
+    if (hasCorrectStatusCodeForNew14B(licenceEntity) && isLicenseOfCorrectType(licenceEntity)) {
+      expandedText1 = if (isLicenseExpiryDateOnOrAfterTwelveMonths(licenceEntity.licenceExpiryDate)) {
+        replaceExpandedEndDateWithARDOrCRD(licenceEntity, expandedText1)
+      } else {
+        replaceExpandedEndDateWithLicenseExpiryDate(expandedText1, licenceEntity)
+      }
+    }
+    return expandedText1
+  }
+
+  private fun hasCorrectStatusCodeForNew14B(
+    licenceEntity: EntityLicence
+  ) = setOf(IN_PROGRESS, SUBMITTED, APPROVED, VARIATION_IN_PROGRESS, VARIATION_SUBMITTED, VARIATION_APPROVED)
+    .contains(licenceEntity.statusCode)
+
+  private fun hasCorrectStatusCodeForExisting14B(
+    licenceEntity: EntityLicence
+  ) = setOf(IN_PROGRESS, SUBMITTED, APPROVED)
+    .contains(licenceEntity.statusCode)
+
+  private fun isLicenseOfCorrectType(
```

Comment:
shouldn't need to worry about licence type as it's implicit based on where the condition.

Reply:
done